### PR TITLE
Replace the ID of the generated connection profile with the one through edit connection

### LINKED ISSUE #10405: Edit Connection Updating Password Resulted in 2 Connections in Servers View
I had a connection saved in ADS to a server. This password was since updated. Right clicked the server and chose "Edit Connection", which resulted in 2 connections to the same server showing up in the Connections viewlet.

### PR DESCRIPTION
**Pull request by aaomidi**
Fixes #10405

### CODE REVIEW

File: src/sql/workbench/services/connection/browser/connectionDialogService.ts [hunk 1]
```diff
@@ -166,6 +166,10 @@ export class ConnectionDialogService implements IConnectionDialogService {
 				}
 				profile = result.connection;
 
+				if (params.oldProfileId && params.isEditConnection) {
+					profile.id = params.oldProfileId;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/connectionDialogService.ts:170]: Can't you just use this._inputModel.id?

(all badness of the whole storing state in the service aside - it's already there so should be usable without passing around more parameters at least)

Comment (aaomidi) [reply src/sql/workbench/services/connection/browser/connectionDialogService.ts:170]: I'll change it to use that

Comment (aaomidi) [reply src/sql/workbench/services/connection/browser/connectionDialogService.ts:170]: I think in the eventual pattern i'll keep this parameter passing as well.

Comment (Charles-Gagnon) [reply src/sql/workbench/services/connection/browser/connectionDialogService.ts:170]: Alright, fair enough as a temporary thing

**Pull request merged**